pyproject.toml

```diff
@@ -179,107 +167,114 @@ quote-style = "double"
+    "B035",   # Don't use static keys in dict comprehensions.
+    "RUF013", # Prohibit implicit Optionals (PEP 484)
+    "RUF100", # #noqa directive that doesn't flag anything
+    "RUF200", # If ruff fails to parse pyproject.toml...
 ]
 ignore = [
-    "ANN10",  # Type annotations for `self` and `cls`
+    "ANN10", # Type annotations for `self` and `cls`
     #"E203",  # Whitespace before ":"  -- Commented because ruff doesn't currently check E203
-    "E501",  # Line too long (reason: black will automatically fix this for us)
-    "D105",  # Missing docstring in magic method (reason: magic methods already have definitions)
-    "D107",  # Missing docstring in __init__ (reason: documented in class docstring)
-    "D203",  # 1 blank line required before class docstring (reason: pep257 default)
-    "D213",  # Multi-line docstring summary should start at the second line (reason: pep257 default)
-    "D215",  # Section underline is over-indented (reason: pep257 default)
-    "A003",  # Class attribute shadowing built-in (reason: Class attributes don't often get bare references)
+    "E501",   # Line too long (reason: black will automatically fix this for us)
+    "D105",   # Missing docstring in magic method (reason: magic methods already have definitions)
+    "D107",   # Missing docstring in __init__ (reason: documented in class docstring)
+    "D203",   # 1 blank line required before class docstring (reason: pep257 default)
+    "D213",   # Multi-line docstring summary should start at the second line (reason: pep257 default)
+    "D215",   # Section underline is over-indented (reason: pep257 default)
+    "A003",   # Class attribute shadowing built-in (reason: Class attributes don't often get bare references)
     "SIM117", # Use a single `with` statement with multiple contexts instead of nested `with` statements
-              # (reason: this creates long lines that get wrapped and reduces readability)
+    # (reason: this creates long lines that get wrapped and reduces readability)
 
     # Ignored due to conflicts with ruff's formatter:
     # https://docs.astral.sh/ruff/formatter/#conflicting-lint-rules
-    "COM812",  # Missing trailing comma - mostly the same, but marginal differences.
-    "ISC001",  # Single-line implicit string concatenation.
+    "COM812", # Missing trailing comma - mostly the same, but marginal differences.
+    "ISC001", # Single-line implicit string concatenation.
 
     # Ignored due to common usage in current code
-    "TRY003",  # Avoid specifying long messages outside the exception class
+    "TRY003", # Avoid specifying long messages outside the exception class
 ]
 
 [tool.ruff.lint.flake8-annotations]
 allow-star-arg-any = true
 
 [tool.ruff.lint.pydocstyle]
-ignore-decorators = [  # Functions with these decorators don't have to have docstrings.
-    "typing.overload",  # Default configuration
+ignore-decorators = [ # Functions with these decorators don't have to have docstrings.
```

Comment:
these spacing changes are inviting future conflicts aren't they?

Reply:
Probably. I looked into it and found my editor (vscode) formats the file. This is unclear to me yet what rules are applied. Unless I missed something it appears we are not linting/formatting TOML files. I ran the various linters we already have in hope it could deal with the TOML file but without success. What do you think about adding a TOML formatter?

Reply:
I think it's a good idea. We discussed it in the past but I can't remember the outcome (don't think we were against it though, we already lint yamls)

Reply:
As discussed this will be solved by adopting a TOML linter.